creme/utils/shift.py

```diff
@@ -0,0 +1,121 @@
+import collections
+
+
+class Shift:
+    """Shift the observations with the selected period.
+
+    Shift class allows to store and recall at regular intervals values observed in a stream.
+    The method ```get()`` returns the ```n-th - period``` observed entry.
+
+    Parameters:
+        period (int): Number of periods to shift. Must be positive.
+        missing: Handle missing values when there is'nt any reference for a given period.
+
+    Example:
+
+        ::
+
+            # Example 1:
+            >>> from creme import utils
+
+            >>> shift = utils.Shift(period=2, missing=-1)
+
+            >>> X = [
+            ...    1,
+            ...    2,
+            ...    3,
+            ...    4,
+            ...    5,
+            ... ]
+
+            >>> for x in X:
+            ...    print(shift.get())
+            ...    _ = shift.update(x)
+            -1
+            -1
+            1
+            2
+            3
+
+            # Example 2.1:
+            >>> from creme import utils
+            >>> from creme import feature_extraction
+
+            # This hypothetical dataset refers to the number of sales of a product on a daily basis.
```

Comment:
You don't need to add hashtags to add a comment in a doctest, just remove them :) and jump a line.